### ISSUE TITLE
Enhancement: Add preset fields to the exporter inorder to include addditional preset conditions  

### ISSUE THREAD

**Issue by rohit-kumar-j** [yanshil/Fusion2PyBullet#7]
Hi! We could add a few preset joint criteria as shown [here](https://wiki.ros.org/urdf/XML/joint). Of course, it will need manual tuning to get it even close to being realistic (eg. joint-limits, calibration, damping, effort, etc), But I think this could be a good enhancement. I will try to create a pull request.
Example:
![image](https://user-images.githubusercontent.com/37873142/107818464-a9250200-6d9d-11eb-9958-16a361d2684e.png)
We could just input these values directly and the users can modify them at will. We will need to provide a disclaimer in the Readme file tho.

Warm Regards,
Rohit

**Comment by yanshil** [comment]
Thank you for the comments.


* Could you please double check that these criteria in URDF will be used and calculated in PyBullet?
    * What I mean here is: they are valid in [urdf](https://wiki.ros.org/urdf/XML/joint), but PyBullet will not calculate their effect.
    * The [documentation](https://docs.google.com/document/d/10sXEhzFRSnvFcl3XxNGhnD4N2SedqwdAvK3dsihxVUA/edit#heading=h.2ye70wns7io3) might list them out. 

When I was working on one of my project, I did met with one criteria I included in the urdf file but not used by PyBullet. (Well, I don't remember what it was but it's my case..... Also, it had been one year that I didn't use PyBullet so there might be some updates I'm not informed))

**Comment by yanshil** [comment]
A note: (I don't have time to carefully check it through now,  but it might be helpful if you are interested in double check this problem)

* Source code of the PyBullet urdf parser: 
* [UrdfParser.h](https://github.com/bulletphysics/bullet3/blob/afa4fb54505fd071103b8e2e8793c38fd40f6fb6/examples/Importers/ImportURDFDemo/UrdfParser.h)
* [UrdfParser.cpp](https://github.com/bulletphysics/bullet3/blob/afa4fb54505fd071103b8e2e8793c38fd40f6fb6/examples/Importers/ImportURDFDemo/UrdfParser.cpp)

**Comment by yanshil** [comment]
In my quick review, <safety_controller> label is actually not parsed in PyBullet.

For example, parameter `lateral_friction` will be parsed, because the source code have

```
XMLElement* friction_xml = ci->FirstChildElement("lateral_friction");
```

But for these enhancement labels, when I search `k_velocity` and `soft`, there is no appearance. It means that they will not be parsed to PyBullet engine and get calculated

**Comment by rohit-kumar-j** [comment]
Thank you for the reply! @yanshil 
       Yes, you are correct. `friction` parameters, `spring` parameters, etc. have been given, but not the `<safety_controller>` I will edit the commit to exclude these attributes. `<limit>`, `<calibration>`, `<dynamics>`, and `<axis>` are included and I will keep them as is.

![image](https://user-images.githubusercontent.com/37873142/109026201-7f48d500-76e5-11eb-9ed2-4f496717d72c.png)


Warm Regards,
Rohit

**Comment by yanshil** [comment]
1. `calibration` is not included in the urdf parser source code[UrdfParse.cpp](https://github.com/bulletphysics/bullet3/blob/afa4fb54505fd071103b8e2e8793c38fd40f6fb6/examples/Importers/ImportURDFDemo/UrdfParser.cpp).

2. `limits` actually do not need to be changed. The current version have  grab limits for prismatic and revolute joint. The only thing you changed is setting the default value  'effort=30' instead of 100
```
limit.attrib = {'upper': str(self.upper_limit), 'lower': str(self.lower_limit),
                'effort': '100', 'velocity': '100'}
```

3. For `dynamics`
* `dampings`: I didn't export them. PyBullet did suggest hard coded `dampings` to 0 before simulation. Setting `dampings` as 0 might  be fine. (In Pybullet Quickstart Guide - Support, Tips, Citation, the 6th Q&A)
* `friction`: But.... friction=0... I don't think it is reasonable to make it default as 0 (If I have got it right, Pybullet made the default as 0.5 ). 

In summary, I think the changes are not necessary to be applied.
